sdk/core/azure-core/test/ut/uuid_test.cpp

```diff
@@ -127,3 +127,49 @@ TEST(Uuid, validChars)
       uuidKey,
       4);
 }
+
+TEST(Uuid, nilAndDefault)
+{
+  Uuid uuid;
+  ASSERT_TRUE(uuid.IsNil());
+  ASSERT_EQ(uuid.ToString(), "00000000-0000-0000-0000-000000000000");
+  ASSERT_EQ(uuid, Uuid{});
+  ASSERT_EQ(uuid.AsArray(), Uuid::ValueArray({0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0}));
+}
+
+TEST(Uuid, parse)
+{
+  Uuid uuid1 = Uuid::Parse("00112233-4455-6677-8899-aAbBcCdDeEfF");
+
+  ASSERT_FALSE(uuid1.IsNil());
+  ASSERT_EQ(uuid1.ToString(), "00112233-4455-6677-8899-aabbccddeeff");
+  ASSERT_NE(uuid1, Uuid{});
+  ASSERT_EQ(
+      uuid1.AsArray(),
+      Uuid::ValueArray(
+          {0x00,
+           0x11,
+           0x22,
+           0x33,
+           0x44,
+           0x55,
+           0x66,
+           0x77,
+           0x88,
+           0x99,
+           0xAA,
+           0xBB,
+           0xCC,
+           0xDD,
+           0xEE,
+           0xFF}));
+
+  ASSERT_THROW(Uuid::Parse("00000000-0000-0000-0000-000000000000 "), std::invalid_argument);
+  ASSERT_THROW(Uuid::Parse("00000000-0000-0000-0000-00000000000"), std::invalid_argument);
+  ASSERT_THROW(Uuid::Parse("00000000a0000-0000-0000-000000000000"), std::invalid_argument);
```

Comment:
Add variations for all the edge cases. Also would be good to test upper case, lower case, symbols, white space ....

```suggestion
  ASSERT_THROW(Uuid::Parse("00000000a0000-0000-0000-000000000000"), std::invalid_argument);
  ASSERT_THROW(Uuid::Parse("00000000-0000a0000-0000-000000000000"), std::invalid_argument);
  ASSERT_THROW(Uuid::Parse("00000000-0000-0000a0000-000000000000"), std::invalid_argument);
  ASSERT_THROW(Uuid::Parse("00000000-0000-0000-0000a000000000000"), std::invalid_argument);
  ASSERT_THROW(Uuid::Parse("00000000000000000000000000000000"), std::invalid_argument);
  ASSERT_THROW(Uuid::Parse("00000000-000000000000000000000000"), std::invalid_argument);
  ASSERT_THROW(Uuid::Parse("00000000-0000-00000000000000000000"), std::invalid_argument);
  ASSERT_THROW(Uuid::Parse("00000000-0000-0000-0000000000000000"), std::invalid_argument);
  ASSERT_THROW(Uuid::Parse("00000000-0000-0000-0000-0000000000G"), std::invalid_argument);
```

Reply:
Ok, I'll add more - for whitespace and random characters, for example. As far as upper/lowercase, there is a test (parsing `"00112233-4455-6677-8899-aAbBcCdDeEfF"`).